src/utilities.jl

```diff
@@ -308,45 +336,35 @@ function printmethod(buffer::IOBuffer, binding::Docs.Binding, func, method::Meth
         typ
     end
 
-    for (i, sym) in enumerate(args)
-        if typesig isa UnionAll
-            # e.g. Tuple{Vector{T}} where T<:Number
-            # or   Tuple{String, T, T} where T<:Number
-            # or   Tuple{Type{T}, String, Union{Nothing, Function}} where T<:Number
-            t = [x for x in f(typesig).types]
-            t = [get_typesig(x, x) for x in t][i]
-        else
-            # e.g. Tuple{Vector{Int}}
-            t = typesig.types[i]
-        end
+    # if `typesig` is an UnionAll, it may be
+    # e.g. Tuple{Vector{T}} where T<:Number
+    # or   Tuple{String, T, T} where T<:Number
+    # or   Tuple{Type{T}, String, Union{Nothing, Function}} where T<:Number
+    # in the other case, it's usually something like Tuple{Vector{Int}}.
+    argtypes = typesig isa UnionAll ?
+            [get_typesig(t, t) for t in find_inner_tuple_type(typesig).types] :
+            collect(typesig.types)
+
+    args = map(args, argtypes) do arg,t
+        type = ""
+        suffix = ""
         if isvarargtype(t)
-            elt = vararg_eltype(t)
-            if elt === Any
-                print(buffer, "$sym...")
-            else
-                print(buffer, "$sym::$elt...")
-            end
-        elseif t === Any
-            print(buffer, sym)
-        else
-            print(buffer, "$sym::$t")
+            t = vararg_eltype(t)
+            suffix = "..."
         end
-
-        if i != length(args)
-            print(buffer, ", ")
+        if t!==Any
+            type = "::$t"
         end
+
+        string(arg)*type*suffix
```

Comment:
```suggestion
        "$arg$type$suffix"
```

or

```suggestion
        string(arg, type, suffix)
```

rather than mixing `string` and `*` here.